stan/math/rev/functor/finite_diff_of_grads_hessian.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef STAN_MATH_REV_FUNCTOR_HESSIAN_HPP
+#define STAN_MATH_REV_FUNCTOR_HESSIAN_HPP
+
+#include <stan/math/rev/meta.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/rev/functor.hpp>
+#include <stan/math/prim/fun/finite_diff_stepsize.hpp>
+#include <stdexcept>
+
+namespace stan {
+namespace math {
+
+/**
+ * Calculate the value and the Hessian of the specified function at
+ * the specified argument using first-order finite difference of gradients,
```

Comment:
The reference says it's 2nd order

Reply:
Too many sense of order here.  Gradients are first order derivatives and Hessians second order, but I think @bbbales2 is right here in that it's a second-order finite-differencing scheme (meaning not just one other point being evaluated).  The reference doc says this:

> Central difference approximation ... Second-order derivatives based on gradient: 2n additional gradient calls are needed:

I think we can stick to using "central difference" rather than trying to quantify the order of finite differencing.  It'd be nice to indicate the number of gradient calls required.  Now that I see this, I realize it should also be more efficient than what we were doing before, which is great, because this and the third-order tests are slow as currently written.